pkg/synthetictests/resource_growth.go

```diff
@@ -123,3 +109,42 @@ func comparePostUpgradeResourceCountFromMetrics(testName, resource string) []*ju
 	}
 
 }
+
+func getAndCheckClusterVersion(testName string) (*v1.ClusterVersion, []*junitapi.JUnitTestCase) {
+	oc := exutil.NewCLI("resource-growth-test")
+	cv, err := oc.AdminConfigClient().ConfigV1().ClusterVersions().Get(context.Background(), "version",
+		metav1.GetOptions{})
+
+	if err != nil {
+		return cv, []*junitapi.JUnitTestCase{
+			{
+				Name: testName,
+				FailureOutput: &junitapi.FailureOutput{
+					Output: "Error getting ClusterVersion: " + err.Error(),
+				},
+			},
+		}
+
+	}
+	if len(cv.Status.History) == 0 {
+		return cv, []*junitapi.JUnitTestCase{
+			{
+				Name: testName,
+				FailureOutput: &junitapi.FailureOutput{
+					Output: "ClusterVersion.Status has no History",
+				},
+			},
+		}
+	}
+
+	// In the case that cv completiontime is nil meaning the version change (upgrade or
+	// rollback is still in progress), skip this case. The problem is likely a bigger
+	// problem and the job will fail for that.
+	if cv.Status.History[0].CompletionTime == nil {
+		return cv, []*junitapi.JUnitTestCase{
+			{Name: "[Skipped] " + testName},
```

Comment:
I believe that modifying the test name in specific cases like this doesn't actually skip anything, and now has created a new test that only runs on rollback jobs, and the other original test only runs on the others. Pass rate tracking is now disjoint between the two in all our tooling. I've seen the [Skipped] syntax and I'm not super familiar with it, but from what I understand it's constant in the test name, and always skipped, or includes a condition that something internal determines is true or not, which I'm guessing we don't want to plumb through for rollback jobs.

I think the more correct thing to do in a synthetic test would be to flake it, keep the original test name, in the slice of junit test results return two entries, fail the first and say why, and pass the second. The test will now appear as a flake in the spyglass UI, sippy, and bigquery.

Reply:
@dgoodwin , please take another look. I've updated it to make it flake now.